File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Models/ProductModels.swift
import Foundation

public struct TrustBadge: Codable, Hashable, Identifiable {
    public let id: String
    public let name: String
    public let icon: String

    public init(id: String, name: String, icon: String) {
        self.id = id
        self.name = name
        self.icon = icon
    }
}

public struct ProductOffer: Codable, Hashable {
    public let type: String
    public let title: String
    public let description: String
    public let discountAmount: Double
    public let code: String?

    public init(
        type: String, title: String, description: String, discountAmount: Double, code: String?
    ) {
        self.type = type
        self.title = title
        self.description = description
        self.discountAmount = discountAmount
        self.code = code
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/data/model/ProductModels.kt
package com.localforvocalstartup.app.data.model

import com.google.gson.annotations.SerializedName

data class TrustBadge(
    val id: String? = null,
    val name: String,
    val icon: String,
    val description: String? = null
)

data class ProductOffer(
    @SerializedName("_id") val id: String? = null,
    val type: String? = null,
    val title: String,
    val description: String? = null,
    val discountAmount: Double? = null,
    val discountType: String? = null,
    val discountValue: Double? = null,
    val code: String? = null
)
